pages/api/project/index.ts

```diff
@@ -93,21 +100,29 @@ async function addProject(args: {
   description: string;
   githubRepository: string;
   tags: string[];
+  coverImg: string;
   authorId: string;
 }) {
-  const { title, description, githubRepository, tags, authorId } = args;
+  const { title, description, githubRepository, tags, coverImg, authorId } =
+    args;
   try {
     const data = await prisma.project.create({
       data: {
         title,
         description,
         githubRepository,
         tags,
-        authorId,
+        coverImg,
+        author: {
+          connect: {
+            id: authorId,
+          },
```

Comment:
could you explain this ? just curious to know

Reply:
@leewan-09

Reply:
Before, if we passed a valid **MongoDB id**, that would work, but it wouldn't check if an **author** with that **id exists**. When we try to fetch that **project**, **Prisma** will **complain** that there is no author **associated** with that project.

However, the **current** approach will **solve** that problem. Before inserting data Prisma, it will check if an **author** with that **ID** already exists in the database, and if not, it will send us an error.

Reply:
great! thanks for the info!